### DISCUSSION TITLE
Super-capf: `cape-dabbrev` duplicates with `elisp-completion-at-point`

### DISCUSSION THREAD

**Discussion by governos** [minad/cape#130] [Q&A]
Snippet:
```emacs-lisp
    (defun my-cape-ignore-elisp-keywords (cand)
      (or (not (keywordp cand))
          (eq (char-after (car completion-in-region--data)) ?:)))

  (defun my-cape-setup-elisp ()
    (setq-local completion-at-point-functions
		`(,(cape-capf-properties
		    (cape-capf-super
                     (cape-capf-predicate
                      #'elisp-completion-at-point
                      #'my-cape-ignore-elisp-keywords)
		     :with
                     #'cape-dabbrev)
		    :sort t))
		  cape-dabbrev-min-length 5))
     
    (add-hook! '(emacs-lisp-mode-hook
   	       eval-expression-minibuffer-setup-hook)
  	     #'my-cape-setup-elisp)
```

Value of `completion-at-point-functions`:
```emacs-lisp
Value in #<buffer modeline-config.el>
(#[0 "\303\302\301\300#\207"
    [(:sort t)
     #[0 "\303\302\301\300#\207"
	 [(:with cape-dabbrev)
	  #[0 "\303\302\301\300#\207"
	      [(my-cape-ignore-elisp-keywords)
	       elisp-completion-at-point cape-wrap-predicate apply]
	      4]
	  cape-wrap-super apply]
	 4]
     cape-wrap-properties apply]
    4])
```

[Adapted from corfu's wiki](https://github.com/minad/corfu/wiki#using-cape-to-tweak-and-combine-capfs)

Hello.
I looked at [Duplicates with cape-capf-super](https://github.com/minad/cape/discussions/104) but it involves `capf-yasnippet` so it did not seem relevant.

I'm taking advantage of `cape-super-capf` to join `elisp-completion-at-point` and `cape-dabbrev`. This is nice as I get unknown symbols before exhausting elisp completions.
I do get duplicates, such as this:
![image](https://github.com/user-attachments/assets/902b094b-6f21-4573-a257-159cda2eb450)

I would rather not have duplicates from `cape-dabbrev` to make room for more completions. I would like for `elisp-completions-at-point` to take precedence whenever there are duplicate completions.

> GNU Emacs 31.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.42, cairo version 1.18.0) of 2024-07-25
> Cape Version: 1.7, commit `75b7cf7`

**Comment by minad** [comment]
> I looked at https://github.com/minad/cape/discussions/104 but it involves capf-yasnippet so it did not seem relevant.

No, everything there is relevant. See in particular https://github.com/minad/cape/discussions/104#discussioncomment-8177729. Right now there is no mechanism to force deduplication if candidates are not exactly equal (equal string and equal text properties).

**Comment by governos** [comment]
Aha, I see, I had misconstrued the properties as being something special, not something that every character has. The Elisp manual has elucidated me, thank you for showing the way.
